store/src/file_sink.rs

```diff
@@ -295,3 +295,64 @@ impl FileSink {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use async_compression::tokio::bufread::GzipDecoder;
+    use futures::stream::StreamExt;
+    use tokio::{fs::DirEntry, io::BufReader};
+    use tokio_util::codec::{length_delimited::LengthDelimitedCodec, FramedRead};
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn writes_a_framed_gzip_encoded_file() {
+        let tmp_dir = tempdir::TempDir::new("entropy_tests").unwrap();
+        let (shutdown_trigger, shutdown_listener) = triggered::trigger();
+        let (sender, receiver) = message_channel(10);
+
+        let mut file_sink = FileSinkBuilder::new(FileType::Entropy, &tmp_dir.path(), receiver)
+            .roll_time(chrono::Duration::milliseconds(100))
+            .create()
+            .await
+            .unwrap();
+
+        let sink_thread = tokio::spawn(async move {
+            let _ = file_sink.run(&shutdown_listener).await;
+        });
+
+        let _ = sender
+            .send(String::into_bytes("hello".to_string()))
+            .await
+            .unwrap();
+
+        std::thread::sleep(time::Duration::from_millis(200));
```

Comment:
Do not std::thread::sleep but use `tokio::time::sleep` with `.await`. async systems don't like you blocking the working thread

Reply:
Fixed in dc8b27e